docs/src/content/ignition/docs/guides/viem.md

```diff
@@ -0,0 +1,149 @@
+
+The `ignition` object exposes a `deploy` method, which takes an Ignition module, deploys it against the current network (in tests this will be the in-memory Hardhat network), and returns the results of the module as typed Viem contract instances.
+
+The `deploy` method takes the Module as its first argument:
+
+```typescript
+const ApolloModule = buildModule("Apollo", (m) => {
+  const apollo = m.contract("Rocket", ["Saturn V"]);
+
+  return { apollo };
+});
+
+it("should have named the rocket Saturn V", async function () {
+  const { apollo } = await hre.ignition.deploy(ApolloModule);
+
+  assert.equal(await apollo.read.name(), "Saturn V");
+});
+```
+
+The `ignition.deploy` method returns an object with a Viem contract instance per `Future` returned as a result in the module.
+
+## Usage with Artifacts
+
+To infer the type of a Viem contract instance, the full type of the ABI must be available when initializing a `Future` within a module.
+
+For named Hardhat contracts, the ABI and type information will be retrieved automatically using [hardhat-viem's type generation](https://hardhat.org/hardhat-runner/docs/advanced/using-viem#contract-type-generation).
+
+If you are passing an artifact object to initialize a `Future` within a module, you will need to ensure the artifact's ABI is either declared with [const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) or defined inline:
+
+```typescript
+const counterArtifact = {
+  // ...
+  abi: [...] as const, // <--- const assertion
+}
+
+const CounterModule = buildModule("Counter", (m) => {
+  const counter = m.contract("Counter", CounterArtifact, [42]);
+
+  return { counter };
+});
+```
+
+If type inference isn't working for the returned contract, it is likely that the ABI type needs a `const` assertion or to be defined inline.
+
+This is because the ABI type must be as narrow as possible for Viem to infer the correct type for the contract, and by default Typescript will broaden the types within object literals.
+
+:::tip
+
+Typescript doesn't support importing JSON as const, it will instead automatically broaden the type, breaking Viem's inference. The artifact must be defined inline within Typescript.
+
+:::
+
+See [Viem's type inference requirements](https://viem.sh/docs/typescript.html#type-inference) for more details.
+
+## Sending transactions with a different account
+
+The `ignition.deploy` method will default to using the first account in Hardhat network's `accounts` array as the sender for all transactions.
+
+You can change this by passing a `defaultSender` within the options object as a second argument to the `deploy` method:
+
+```typescript
+const [first, second] = await hre.viem.getWalletClients();
+
+const result = await hre.ignition.deploy(ApolloModule, {
+  defaultSender: second.account.address,
+});
+```
+
+---
```

Comment:
Let's add a title instead of `---` here.

Reply:
I switched it to `Hardhat Ignition and Hardhat Viem`.